test/unit/math/mix/mat/fun/inverse_spd_test.cpp

```diff
@@ -1,305 +1,53 @@
-  stan::math::matrix_ffv m3 = inverse_spd(m1);
-
-  for (int i = 0; i < 3; i++) {
-    for (int j = 0; j < 3; j++) {
-      EXPECT_FLOAT_EQ(m2(i, j).val_.val_.val(), m3(i, j).val_.val_.val());
-      EXPECT_FLOAT_EQ(m2(i, j).d_.val_.val(), m3(i, j).d_.val_.val());
-    }
-  }
-
-  std::vector<var> z1;
-  z1.push_back(m1(0, 0).val_.val_);
-  z1.push_back(m1(0, 1).val_.val_);
-  z1.push_back(m1(0, 2).val_.val_);
-  z1.push_back(m1(1, 0).val_.val_);
-  z1.push_back(m1(1, 1).val_.val_);
-  z1.push_back(m1(1, 2).val_.val_);
-  z1.push_back(m1(2, 0).val_.val_);
-  z1.push_back(m1(2, 1).val_.val_);
-  z1.push_back(m1(2, 2).val_.val_);
-
-  for (int i = 0; i < 3; i++) {
-    for (int j = 0; j < 3; j++) {
-      VEC h1;
-      VEC h2;
-      m2(i, j).d_.val_.grad(z1, h1);
-      stan::math::recover_memory();
-      m3(i, j).d_.val_.grad(z1, h2);
-      stan::math::recover_memory();
-      for (int k = 0; k < 9; k++)
-        EXPECT_FLOAT_EQ(h1[k], h2[k]);
-    }
-  }
-}
-
-TEST_F(AgradMixMatrixInverseSPD, matrix_ffv_3rd_deriv) {
-  using stan::math::inverse_spd;
-
-  stan::math::matrix_ffv m1(3, 3);
-  m1 << 2, -1, 0, -1, 2, -1, 0, -1, 2;
-  m1(0, 0).d_ = 1.0;
-  m1(0, 1).d_ = 1.0;
-  m1(0, 2).d_ = 1.0;
-  m1(1, 0).d_ = 1.0;
-  m1(1, 1).d_ = 1.0;
-  m1(1, 2).d_ = 1.0;
-  m1(2, 0).d_ = 1.0;
-  m1(2, 1).d_ = 1.0;
-  m1(2, 2).d_ = 1.0;
-  m1(0, 0).val_.d_ = 1.0;
-  m1(0, 1).val_.d_ = 1.0;
-  m1(0, 2).val_.d_ = 1.0;
-  m1(1, 0).val_.d_ = 1.0;
-  m1(1, 1).val_.d_ = 1.0;
-  m1(1, 2).val_.d_ = 1.0;
-  m1(2, 0).val_.d_ = 1.0;
-  m1(2, 1).val_.d_ = 1.0;
-  m1(2, 2).val_.d_ = 1.0;
-
-  stan::math::matrix_ffv m2 = stan::math::inverse(m1);
-  stan::math::matrix_ffv m3 = inverse_spd(m1);
-
-  for (int i = 0; i < 3; i++) {
-    for (int j = 0; j < 3; j++) {
-      EXPECT_FLOAT_EQ(m2(i, j).val_.val_.val(), m3(i, j).val_.val_.val());
-      EXPECT_FLOAT_EQ(m2(i, j).d_.val_.val(), m3(i, j).d_.val_.val());
-    }
-  }
-
-  std::vector<var> z1;
-  z1.push_back(m1(0, 0).val_.val_);
-  z1.push_back(m1(0, 1).val_.val_);
-  z1.push_back(m1(0, 2).val_.val_);
-  z1.push_back(m1(1, 0).val_.val_);
-  z1.push_back(m1(1, 1).val_.val_);
-  z1.push_back(m1(1, 2).val_.val_);
-  z1.push_back(m1(2, 0).val_.val_);
-  z1.push_back(m1(2, 1).val_.val_);
-  z1.push_back(m1(2, 2).val_.val_);
-
-  for (int i = 0; i < 3; i++) {
-    for (int j = 0; j < 3; j++) {
-      VEC h1;
-      VEC h2;
-      m2(i, j).d_.d_.grad(z1, h1);
-      stan::math::recover_memory();
-      m3(i, j).d_.d_.grad(z1, h2);
-      stan::math::recover_memory();
-      for (int k = 0; k < 9; k++)
-        EXPECT_FLOAT_EQ(h1[k], h2[k]);
-    }
-  }
+#include <test/unit/math/test_ad.hpp>
+
+TEST(MathMixMatFun, inverseSpd) {
+  auto f = [](const auto& x) {
+    if (x.rows() != x.cols())
+      return stan::math::inverse_spd(x);
+    auto y = ((x + x.transpose()) * 0.5).eval();  // symmetry for finite diffs
+    return stan::math::inverse_spd(y);
+  };
+
+  Eigen::MatrixXd a(2, 2);
```

Comment:
Zero-sized input test?

Reply:
added size 1 (works)

filed issue to get errors in line for size 0: https://github.com/stan-dev/math/issues/1432